controllers/propagator/policy_controller.go

```diff
@@ -122,20 +123,63 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return reconcile.Result{}, err
 			}
 
+			if len(replicatedPlcList.Items) == 0 {
+				log.V(2).Info("No replicated policies to delete.")
+
+				return reconcile.Result{}, nil
+			}
+
+			policiesChan := make(chan policiesv1.Policy, len(replicatedPlcList.Items))
+			deletionResultsChan := make(chan deletionResult, len(replicatedPlcList.Items))
+
+			var numWorkers int
+			if len(replicatedPlcList.Items) > concurrencyPerPolicy {
+				numWorkers = concurrencyPerPolicy
+			} else {
+				numWorkers = len(replicatedPlcList.Items)
+			}
+
+			for i := 0; i < numWorkers; i++ {
+				go plcDeletionWrapper(r, policiesChan, deletionResultsChan)
+			}
+
+			log.V(2).Info("Scheduling work to handle deleting replicated policies for the policy", "policy",
```

Comment:
The key/value pair is duplicated from the `log.WithValues` call at the beginning of the method. My suggestion is to remove it here.

You could optionally rename the key/value pairs in `log.WithValues` to clearly indicate it is the policy name and namespace.